lib/client.js

```diff
@@ -64,8 +64,9 @@ function _connect (client) {
 
   client[kQueue].pause()
 
-  let body = null
-  let read = 0
+  let _body = null
+  let _read = 0
+  let _callback = null
```

Comment:
using _ to indicate state vars to avoid shadowing local vars

Reply:
I'm not really a fan. I think we might look into rewriting/simplifying this whole method. Maybe we can use a state object somewhere and avoid keeping the context in the closures.

Anyway, this is ok for now.

Reply:
I'm happy to revisit this later.